habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -290,9 +485,39 @@ def get_all_scenedataset_receptacles(sim) -> Dict[str, Dict[str, List[str]]]:
     return receptacles
 
 
+def import_tri_mesh_ply(ply_file: str) -> Tuple[List[mn.Vector3], List[int]]:
+    """
+    Returns a Tuple of (verts,indices) from a ply mesh using magnum trade importer.
+
+    :param ply_file: The input PLY mesh file. NOTE: must contain only triangles.
+    """
+    manager = mn.trade.ImporterManager()
+    importer = manager.load_and_instantiate("AnySceneImporter")
+    importer.open_file(ply_file)
+
+    # NOTE: We don't support mesh merging or multi-mesh parsing currently
+    if importer.mesh_count > 1:
+        raise NotImplementedError("TODO: multi-mesh receptacle support.")
```

Comment:
```suggestion
    # TODO: We don't support mesh merging or multi-mesh parsing currently
    if importer.mesh_count > 1:
        raise NotImplementedError("Importing multi-mesh receptacles (mesh merging or multi-mesh parsing) is not supported. ")
```

Reply:
I could expose [MeshTools::concatenate()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1MeshTools.html#a3f38d4e79ecbb0d41f4d58a1f5a59c48) and [SceneTools::flattenMeshHierarchy()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1SceneTools.html#a8b90ec6a17a8f3f3d1150dfdcedab40c) for doing this if you want.

Reply:
I think folks are hoping to eventually have multiple separate meshes in a single file which we'll split out here.
Generally happy to see more Magnum mesh tools in python though. :slightly_smiling_face:

Reply:
Update: I think we want to store separate receptacle meshes as glTF nodes. So here we'll load the glTF file and iterate over all meshes in the scene, creating a TriangleMeshReceptacle for each one.